pkg/splunk/enterprise/indexercluster.go

```diff
@@ -73,6 +75,15 @@ func ApplyIndexerCluster(client splcommon.ControllerClient, cr *enterprisev1.Ind
 		return result, err
 	}
 
+	mgr := indexerClusterPodManager{log: scopedLog, cr: cr, secrets: namespaceScopedSecret, newSplunkClient: splclient.NewSplunkClient}
```

Comment:
Why did we have to add/move the code block to before the `DeletionTimestamp` check?

Reply:
Because we want to check the validity of the spec in the beginning and you need `mgr` for that.